Google.PowerShell/Compute/GceManagedInstanceGroupCmdlets.cs

```diff
@@ -910,42 +919,67 @@ private class ParameterSetNames
 
         protected override void ProcessRecord()
         {
+            bool regionSpecified = MyInvocation.BoundParameters.ContainsKey(nameof(Region));
+            bool zoneSpecified = MyInvocation.BoundParameters.ContainsKey(nameof(Zone));
+            if (regionSpecified && zoneSpecified)
+            {
+                throw new PSInvalidOperationException(
+                    "Parameters -Region and -Zone cannot be used together.");
+            }
+
+            if (InstanceObject != null)
+            {
+                InstanceUri = InstanceObject.Select(instance => instance.SelfLink).ToArray();
+            }
+
             switch (ParameterSetName)
             {
                 case ParameterSetNames.AbandonUri:
-                    AbandonUri();
-                    break;
                 case ParameterSetNames.AbandonObject:
-                    AbandonObject();
+                    AbandonUri(regionSpecified);
                     break;
                 case ParameterSetNames.DeleteUri:
-                    DeleteUri();
-                    break;
                 case ParameterSetNames.DeleteObject:
-                    DeleteObject();
+                    DeleteUri(regionSpecified);
                     break;
                 case ParameterSetNames.RecreateUri:
-                    RecreateUri();
-                    break;
                 case ParameterSetNames.RecreateObject:
-                    RecreateObject();
+                    RecreateUri(regionSpecified);
                     break;
                 case ParameterSetNames.Resize:
-                    Resize();
+                    Resize(regionSpecified);
                     break;
                 case ParameterSetNames.SetTemplate:
-                    SetTemplate();
+                    SetTemplate(regionSpecified);
                     break;
                 case ParameterSetNames.SetTargetPools:
-                    SetTargetPools();
+                    SetTargetPools(regionSpecified);
                     break;
                 default:
                     throw UnknownParameterSetException;
             }
         }
 
-        private void SetTemplate()
+        private void SetTemplate(bool isRegionSpecified)
```

Comment:
I think it may be worthwhile to keep these functions separate as they don't share any logic. 

Maybe still have SetTemplate and then have SetTemplateZone and SetTemplateRegion?